python/ray/tests/aws/test_autoscaler_aws.py

```diff
@@ -225,17 +226,38 @@ def test_fills_out_amis(iam_client_stub, ec2_client_stub):
     ec2_client_stub.assert_no_pending_responses()
 
 
-def test_temp_create_sg_multinode(iam_client_stub, ec2_client_stub):
+def test_missing_key(iam_client_stub):
+    """
+    Check that an error is raised when ssh_private_key is given in the
+    auth_config but a node type is missing KeyName.
+    """
+    config = helpers.load_aws_example_config_file("example-full.yaml")
+    config["auth"]["ssh_private_key"] = "~/fake/path"
+    config["available_node_types"]["ray.head.default"]["node_config"][
+        "KeyName"] = "fakeKey"
+
+    # bootstrap_config configures iam role and then raises an assertion error
+    stubs.configure_iam_role_default(iam_client_stub)
+    with pytest.raises(AssertionError):
+        helpers.bootstrap_aws_config(config)
+
+    # expect no pending responses left in IAM client stub queue
+    iam_client_stub.assert_no_pending_responses()
+
```

Comment:
Could you add one more test to ensure the `UserData` configuration works correctly? i.e. having the `ssh_private_key` passed in and having `["node_config"]["UserData"]` set to a non-empty value